redfish-core/lib/account_service.hpp

```diff
@@ -126,55 +127,181 @@ inline std::string getPrivilegeFromRoleId(std::string_view role)
     return "";
 }
 
-/**
- * @brief Maps user group names retrieved from D-Bus object to
- * Account Types.
- *
- * @param[in] userGroups List of User groups
- * @param[out] res AccountTypes populated
- *
- * @return true in case of success, false if UserGroups contains
- * invalid group name(s).
- */
-inline bool translateUserGroup(const std::vector<std::string>& userGroups,
-                               crow::Response& res)
+inline bool getAccountTypeFromUserGroup(std::string_view userGroup,
+                                        nlohmann::json& accountTypes)
 {
-    std::vector<std::string> accountTypes;
-    for (const auto& userGroup : userGroups)
+    // set false if userGroup values are not found in list, return error
+    bool isFoundUserGroup = true;
+
+    if (userGroup == "redfish")
+    {
+        accountTypes.push_back("Redfish");
+    }
+    else if (userGroup == "ipmi")
+    {
+        accountTypes.push_back("IPMI");
+    }
+    else if (userGroup == "ssh")
     {
-        if (userGroup == "redfish")
+        accountTypes.push_back("HostConsole");
```

Comment:
@joseph-reynolds do we need a downstream patch to look at that downstream HostConsole group?

Reply:
This is what we had for 1030, we can address later